backend/src/backend/models.py

```diff
@@ -0,0 +1,141 @@
+from datetime import datetime
+from typing import Any, Dict, List, Optional
+from uuid import UUID
+
+from sqlalchemy import DateTime, ForeignKey, String, text
+from sqlalchemy.dialects.postgresql import ARRAY, JSONB
+from sqlalchemy.ext.mutable import MutableDict, MutableList
+from sqlalchemy.orm import (
+    DeclarativeBase,
+    Mapped,
+    MappedAsDataclass,
+    mapped_column,
+    relationship,
+)
+from sqlalchemy.sql.schema import MetaData
+
+
+class Base(MappedAsDataclass, DeclarativeBase):
+    # This map details the specific transformation of types between Python and
+    # PostgreSQL. This is only needed for the case where a specific PostgreSQL
+    # type has to be used or when we want to ensure a specific setting (like the
+    # timezone below)
+    type_annotation_map = {
+        Dict[str, Any]: MutableDict.as_mutable(
+            JSONB
+        ),  # transform Python Dict[str, Any] into PostgreSQL JSONB
+        List[Dict[str, Any]]: MutableList.as_mutable(JSONB),
+        datetime: DateTime(
+            timezone=False
+        ),  # transform Python datetime into PostgreSQL Datetime without timezone
+        List[str]: ARRAY(
+            item_type=String
+        ),  # transform Python List[str] into PostgreSQL Array of strings
+    }
+
+    # This metadata specifies some naming conventions that will be used by
+    # alembic to generate constraints names (indexes, unique constraints, ...)
+    metadata = MetaData(
+        naming_convention={
+            "ix": "ix_%(column_0_label)s",
+            "uq": "uq_%(table_name)s_%(column_0_name)s",
+            "ck": "ck_%(table_name)s_%(constraint_name)s",
+            "fk": "fk_%(table_name)s_%(column_0_name)s_%(referred_table_name)s",
+            "pk": "pk_%(table_name)s",
+        }
+    )
+    pass
+
+
+class User(Base):
+    """
+    User model, used for managing users.
+    """
+
+    __tablename__ = "users"
+
+    id: Mapped[UUID] = mapped_column(
+        init=False, primary_key=True, server_default=text("uuid_generate_v4()")
+    )
+    created_on: Mapped[datetime]
+
+    collections: Mapped[List["Collection"]] = relationship(
+        back_populates="user", cascade="all, delete-orphan", init=False
+    )
+
+
+class Collection(Base):
+    """
+    Collection model, used for managing collections.
+    A collection is a group of files and archives.
+    collection will be deleted after certain time.
+    """
+
+    __tablename__ = "collections"
+
+    id: Mapped[UUID] = mapped_column(
+        init=False, primary_key=True, server_default=text("uuid_generate_v4()")
+    )
+    user_id: Mapped[UUID] = mapped_column(ForeignKey("users.id"), init=False)
+    name: Mapped[str]
+    email: Mapped[Optional[str]]
```

Comment:
I think it makes more sense for it to be on Archive: it's to receive notification about a ZIM so it's linked to the Archive. If one records an email address then hit the ZIM button and before the ZIM is completed changes the value of the email field, I'd still expect to receive the notification as the first email.